dev/WindowsAppSDK_Insights/WindowsAppSDKInsights.h

```diff
@@ -10,8 +10,50 @@
 #error "WIL Tracelogging.h must not be explicitly included when including this file"
 #endif
 
-char* WINDOWSAPPSDK_PACKAGE_VER();
-char* WINDOWSAPPSDK_EXPERIMENTATION_LEVEL();
+#include <wil/resource.h>
+#include <string>
+
+#define _WINDOWSAPPSDK_PACKAGE_VER_RESOURCEID             10000
+#define _WINDOWSAPPSDK_EXPERIMENTATION_FLAG_RESOURCEID    10001
+
+#define INSIGHTSRESOURCEDLL "Microsoft.WindowsAppSDK.Insights.Resource.dll"
+
+inline std::string load_resource_string(const char* dllName, uint32_t id)
+{
+    wil::unique_hmodule module(::LoadLibraryA(dllName));
+
+    if (module)
+    {
+        char buffer[256];
+        if (0 < LoadStringA(module.get(), id, buffer, ARRAYSIZE(buffer)))
+        {
+            return buffer;
+        }
+        else
+        {
+            LOG_LAST_ERROR_MSG("Unable to load resource string, id: %d, error: %d", id, GetLastError());
+        }
+        
+    }
+    else
+    {
+        LOG_LAST_ERROR_IF_MSG(module == nullptr, "Unable to load dll: %s", dllName);
+    }
+
+    return {};
+}
+
+__declspec(dllexport) char* WINDOWSAPPSDK_PACKAGE_VER()
```

Comment:
>__declspec(dllexport) [](http://example.com/codeflow?start=0&length=21)

needs to be inline or static to avoid LNK2005

e.g.:
`Error LNK2005 "char * __cdecl WINDOWSAPPSDK_PACKAGE_VER(void)" (?WINDOWSAPPSDK_PACKAGE_VER@@YAPEADXZ) already defined in LoggingTests.obj MrmBaseUnitTests S:\ProjectReunion\dev\MRTCore\mrt\mrm\UnitTests\mrmmin.lib(MrmTraceLogging.obj) 1 `

Reply:
LNK 2005 means that the function is already defined. That can happen if the file and the lib are included. Are you sure that is not the case. The file itself is wrapped in #ifndef 

The __declspec(dllexport) must go.

Reply:
#ifndef doesn't prevent duplication across different compilation units though, just within one.